Bottom: 7eb5da83a7960ca071a7733dc5f37107ea596679
Top:    02460ffe8e6d70967b3c7e05050fb441421d2cf3
Author: melsdoerfer <michael@elsdoerfer.com>
Date:   2008-12-21 16:23:15 +0100

No longer convert an item id (guid) into an absolute link if isPermaLink=="false".

While not strictly necessary to run FeedPlatform, it is strongly recommended. For example, one significant issue arising from this is that the domain name used to build the absolute url may change (say the feed is redirected to a new address), thus effectively changing all the ids.

Reported as issue #107.


---

diff --git a/feedparser/feedparser.py b/feedparser/feedparser.py
index 0a1959b..bfebd8e 100644
--- a/feedparser/feedparser.py
+++ b/feedparser/feedparser.py
@@ -732,7 +732,9 @@ class _FeedParserMixin:
                 
         # resolve relative URIs
         if (element in self.can_be_relative_uri) and output:
-            output = self.resolveURI(output)
+            # but not for isPermaLink=False guids in rss
+            if not (element == 'id' and self.version.startswith('rss') and self.inentry and not self.guidislink):
+                output = self.resolveURI(output)
         
         # decode entities within embedded markup
         if not self.contentparams.get('base64', 0):
